rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -0,0 +1,352 @@
+        and the unbond requests from its bonds.
+        """
+        address_bonds = defaultdict(list)
+        address_unbonds_set = {(unbond.address, unbond.bond_id) for unbond in unbonds}
+        address_unbond_requests_set = {
+            (unbond_request.address, unbond_request.bond_id) for unbond_request in unbond_requests
+        }
+        # Get bonds that have not been either unbonded or requested to be unbond
+        for bond in bonds:
+            if (
+                (bond.address, bond.bond_id) in
+                address_unbonds_set.union(address_unbond_requests_set)
+            ):
+                continue
+            address_bonds[bond.address].append(bond)
+
+        # Get per address staked balances in pools
+        adex_balances: DefaultDict[ChecksumAddress, List[PoolBalance]] = defaultdict(list)
+        for address, bonds in address_bonds.items():
+            pool_ids = {bond.pool_id for bond in bonds}
+            for pool_id in pool_ids:
+                amount = FVal(sum(bond.amount for bond in bonds if bond.pool_id == pool_id))
+                pool_balance = PoolBalance(
+                    pool_id=pool_id,
+                    pool_name=POOL_ID_POOL_NAME[pool_id],
+                    token=EthereumToken('ADX'),
+                    balance=Balance(
+                        amount=amount,
+                        usd_value=Price(amount) * adx_usd_price,
+                    ),
+                )
+                adex_balances[address].append(pool_balance)
+
+        return dict(adex_balances)
+
+    @overload  # noqa: F811
+    def _deserialize_event_graph(  # pylint: disable=no-self-use
+            self,
+            raw_event: Dict[str, Any],
+            case: Literal['bond'],
+            identity_address_map: Dict[ChecksumAddress, ChecksumAddress],
+    ) -> Bond:
+        ...
+
+    @overload  # noqa: F811
+    def _deserialize_event_graph(  # pylint: disable=no-self-use
+            self,
+            raw_event: Dict[str, Any],
+            case: Literal['unbond'],
+            identity_address_map: Dict[ChecksumAddress, ChecksumAddress],
+    ) -> Unbond:
+        ...
+
+    @overload  # noqa: F811
+    def _deserialize_event_graph(  # pylint: disable=no-self-use
+            self,
+            raw_event: Dict[str, Any],
+            case: Literal['unbond_request'],
+            identity_address_map: Dict[ChecksumAddress, ChecksumAddress],
+    ) -> UnbondRequest:
+        ...
+
+    def _deserialize_event_graph(  # pylint: disable=no-self-use
+            self,
+            raw_event: Dict[str, Any],
+            case: Literal['bond', 'unbond', 'unbond_request'],
+            identity_address_map: Dict[ChecksumAddress, ChecksumAddress],
+    ) -> Union[Bond, Unbond, UnbondRequest]:
+        """Given a subgraph event entry returns the right event instance.
+
+        NB: this method should be an staticmethod, but at the moment
```

Comment:
If it's a mypy bug then just use `type:ignore` and make it a staticmethod. Or ... even better take it out of the class, make it a normal function and there overload works for sure (we have it in other parts of the code)

Don't write wrong code to satisfy mypy bugs.